src/ngx_http_lua_sleep.c

```diff
@@ -36,23 +36,34 @@ ngx_http_lua_ngx_sleep(lua_State *L)
         return luaL_error(L, "attempt to pass %d arguments, but accepted 1", n);
     }
 
-    r = ngx_http_lua_get_req(L);
-    if (r == NULL) {
-        return luaL_error(L, "no request found");
-    }
-
     delay = (ngx_int_t) (luaL_checknumber(L, 1) * 1000);
 
     if (delay < 0) {
         return luaL_error(L, "invalid sleep duration \"%d\"", delay);
     }
 
+    r = ngx_http_lua_get_req(L);
+    if (r == NULL) {
+        /* init_by_lua phase */
+        ngx_log_debug1(NGX_LOG_DEBUG_HTTP, ngx_cycle->log, 0,
```

Comment:
should we have a different log format for sleep in non NGX_HTTP_LUA_CONTEXT_YIELDABLE? Or we could log with phase in the debug log?